actions/supr.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+
+import requests
+import json
+from datetime import date
+from dateutil.relativedelta import relativedelta
+
+# Needs to be run in a Stackstorm virtualenv
+from st2actions.runners.pythonrunner import Action
+
+
+class Supr(Action):
+
+    DATE_FORMAT = '%Y-%m-%d'
+
+    @staticmethod
+    def search_by_email(base_url, email, user, key):
+        search_person_url = '{}/person/search/'.format(base_url)
+        # Search case insensitive
+        params = {'email_i': email}
+        response = requests.get(search_person_url, params=params, auth=(user, key))
+
+        if response.status_code != 200:
+            raise AssertionError("Status code returned when trying to get PI id for email: "
+                                 "{} was not 200.".format(email))
+
+        response_as_json = json.loads(response.content)
+        matches = response_as_json["matches"]
+
+        if len(matches) < 1:
+            raise AssertionError("There were no hits in SUPR for email: {}".format(email))
+
+        if len(matches) > 1:
+            raise AssertionError("There we more than one hit in SUPR for email: {}".format(email))
+
+        return matches[0]["id"]
+
+    @staticmethod
+    def search_for_pis(project_to_email_dict, supr_base_api_url, api_user, api_key):
+        res = {}
+        for project, email in project_to_email_dict.iteritems():
+            res[project] = Supr.search_by_email(base_url=supr_base_api_url,
+                                                email=email,
+                                                user=api_user,
+                                                key=api_key)
+        return res
+
+    @staticmethod
+    def create_delivery_project(base_url, ngi_project_name, pi_id, user, key):
+
+        create_delivery_project_url = '{}/ngi_delivery/project/create/'.format(base_url)
+
+        today = date.today()
+        today_formatted = today.strftime(Supr.DATE_FORMAT)
+        six_months_from_now = today + relativedelta(months=+6)
+        six_months_from_now_formatted = six_months_from_now.strftime(Supr.DATE_FORMAT)
+
+        payload = {
+            'ngi_project_name': ngi_project_name,
+            'title': "DELIVERY_{}_{}".format(ngi_project_name, today_formatted),
+            'pi_id': pi_id,
+            'start_date': today_formatted,
+            'end_date': six_months_from_now_formatted,
+            'continuation_name': '',
+            # TODO Right now this default to 1T, we might want to get something
+            # realistic in here later.
+            # 'allocated': size_of_delivery,
+            'api_opaque_data': '',
+            'ngi_ready': False,
```

Comment:
When we're calling Mover directly as a shell command then yes, one could argue that Mover can set this itself. But if we want Mover to poll SUPR for ready projects to deliver then we should set this to `True` whenever we want to delegate the responsibility. So perhaps we could set it to `True` just before we call Mover in the mean time?